noir-projects/noir-contracts/contracts/account/simulated_account_contract/src/main.nr

```diff
@@ -0,0 +1,49 @@
+mod public_key_note;
+
+use dep::aztec::macros::aztec;
+
+#[aztec]
+pub contract SimulatedAccount {
+    use dep::authwit::{
+        account::AccountActions,
+        auth::{compute_authwit_message_hash, IS_VALID_SELECTOR},
+        entrypoint::{app::AppPayload, fee::FeePayload},
+    };
+    use dep::aztec::{
+        macros::{functions::{private, view}, storage::storage},
+        oracle::offchain_message::emit_offchain_message,
+        prelude::{PrivateContext, PrivateImmutable},
+    };
+
+    use crate::public_key_note::PublicKeyNote;
+
+    #[storage]
+    struct Storage<Context> {
+        signing_public_key: PrivateImmutable<PublicKeyNote, Context>,
+    }
+
+    // Note: If you globally change the entrypoint signature don't forget to update account_entrypoint.ts
+    #[private]
+    fn entrypoint(app_payload: AppPayload, fee_payload: FeePayload, cancellable: bool) {
+        let actions = AccountActions::init(&mut context, is_valid_impl);
+        actions.entrypoint(app_payload, fee_payload, cancellable);
+    }
+
+    #[private]
+    #[view]
+    fn verify_private_authwit(inner_hash: Field) -> Field {
+        let message_hash = compute_authwit_message_hash(
+            context.msg_sender(),
+            context.chain_id(),
+            context.version(),
+            inner_hash,
+        );
+        emit_offchain_message(message_hash, context.msg_sender());
```

Comment:
This is the important piece to document. Impossible to understand without it

Also is my understanding correct that we currently lump all the offchain messages together and we don't type them in any way? (i.e. we don't differentiate between the messages passed to `process_message` and the authwit messages you emit here)

If yes would create an issue and link it here.